awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -375,7 +379,10 @@ def delete_if_needed(self, existing_item, on_delete=None):
                 else:
                     self.fail_json(msg="Unable to delete {0} {1}: {2}".format(item_type, item_name, response['status_code']))
         else:
-            self.exit_json(**self.json_output)
+            if auto_exit:
+                self.exit_json(**self.json_output)
+            else:
+                return None
```

Comment:
Is there anyway we can standardize on what gets returned if you are doing `auto_exit=False`? We have return self.json_output, None and I think another two options below. If we can't standardize on a return can we just return nothing and then have methods to get at the required data from the TowerAPIModule directly? i.e. for json_output you can reference `module.json_output` from the calling module.

Reply:
changed to return self.json_output , since we are adding things to it as we go, and then can be consistent, unless you were thinking of something different?